site/src/pages/release-notes/1.24.0.mdx

```diff
@@ -0,0 +1,279 @@
+  ```java
+  GraphqlService
+    .builder()
+    .executionIdGenerator((requestCtx, query, operation, graphqlCtx) -> {
+       return ExecutionId.from(requestContext.headers.get("traceparent"));
+    })
+    ...
+    .build();
+  ```
+- **Custom CoroutineContext for gRPC-Kotlin**: You can now inject a custom `CoroutineContext` for gRPC-Kotlin
+  with <type://CoroutinContextProvider>. #4842 #4866
+  ```kotlin
+  class CustomCoroutineContextProvider : CoroutineContextProvider { 
+    override fun provide(ctx: ServiceRequestContext): CoroutineContext { 
+      return ... // A custom CoroutineContext 
+    }  
+  }
+  ``` 
+  <Tip>
+  
+  Note that a custom `CoroutineContextProvider` should be registered via Java SPI to 
+  `META-INF/services/com.linecorp.armeria.common.kotlin.CoroutineContextProvider`.
+ 
+  </Tip>
+- **Retrieving CallOptions with GrpcCallOptions**: With GrpcCallOptions in Armeria, you can now easily retrieve
+  `CallOptions` from a given <type://RequestContext>.
+  #4776 #4822
+  ```java
+  GrpcClients
+    .builder(grpcServerUri)
+    .decorator((delegate, ctx, req) -> {
+      CallOptions options = GrpcCallOptions.get(ctx);
+      MyOption myOption = options.getOption(myOptionKey);
+      // act on myOption if needed
+      ...
+  
+      return delegate.execute(ctx, req);
+    })
+    .build(MyGrpcStub.class);
+  ```
+- **Fallback Strategy for CircuitBreakerClient**: You can now easily set a fallback strategy to 
+  <type://CircuitBreakerClient> when a <type://CircuitBreaker> is open. #4800 #4818
+  ```java
+  CircuitBreakerClient
+    .builder(...)
+    .recover((ctx, req) -> {
+      // A fallback response when the circuit is open.
+      return HttpResponse.of(...);
+    });
+  ```
+- **Netty ChannelPipeline Customizer**: You can now customize the Netty `ChannelPipeline` for the server side #4710 #4813
+  ```java
+  Server
+    .builder()
+    .childChannelPipelineCustomizer(pipeline -> {
+      // Add a custom handler to the pipeline.
+    })
+    .build();
+  ```
+- **HTTP/2 over TLS without ALPN**: Armeria now supports using HTTP/2 over TLS without the need for 
+  [ALPN](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-http2-17#section-3.3) #4702 #4704
+  ```java
+  ClientFactory
+    .builder()
+    .useHttp2WithoutAlpn(true)
+    ...
+    .build();
+  ```
+- **Armeria with Jetty**: Armeria is now compatible with Jetty 10 and Jetty 11 #4845
+- **Armeria with Thrift**: Armeria is now compatible with Thrift 0.18.1. #4812
+
+## 📈 Improvements
+
+- `RequestContext.makeContextAware()` now returns instances of <type://ContextAwareRunnable>,
+  <type://ContextAwareCallable>, <type://ContextAwareFunction>, <type://ContextAwareBiFunction>,
+  <type://ContextAwareConsumer> and <type://ContextAwareBiConsumer>. #4878 #4890
+- Annotated services now perform as fast as functional-style services under the following conditions: #4888
+  - The annotated method returns an `HttpResponse`
```

Comment:
```suggestion
  - The annotated method returns an `HttpResponse`.
```
Would you mind adding dots to the sentences that miss?

Reply:
Good point. I checked all sentences.